### ISSUE TITLE
Using React component classes

### ISSUE THREAD

**Issue by kristianmandrup** [flowhub/the-graph#345]
My [polymer2](https://github.com/kristianmandrup/the-graph/tree/polymer2) branch now has support for using ES6 classes with mixins. 

Feel free to help upgrade the rest of the components to ES6 style

**Comment by jonnor** [comment]
Everything changed, not so easy to review...

* Are there changes in behavior, or is it the same?
* Which features are tested?
* Anything which needs to be done before it could be merged and used in production apps?

**Comment by jonnor** [comment]
Extension of #344

**Comment by kristianmandrup** [comment]
I cloned the repo then pushed. Didn't fork, why commit stories might be out of sync.
I documented most of the changes in a migration doc in root folder. Most everything should work.

**Comment by kristianmandrup** [comment]
No changes in behavior. Been working on network graph example only. Can draw connections between nodes as before.

**Comment by jonnor** [comment]
By migration doc, do you mean 'Polymer2-migration-notes.md'? That file is empty. Anyways,

To you intend for this to branch to merged into this repository?
Do you intend to help maintain it if it gets merged?

If so there are a lot of things that needs to be fixed. But before we go to details, there are two major things that must be dealt with. For this to be at all possible to merge, it needs to:

1) Help to remove Polymer dependency (or at least not make it worse)
2) Not break existing functionality

# 1
This branch seems to add *more* dependencies on Polymer. And furthermore changes the version to a yet-to-be-released version. Do the React components still work with old Polymer? Can the components be used without Polymer? Does it get us closer to working without Polymer somehow?

# 2
I tested the `examples/demo-network.html` (in latest Chrome). Several pieces of basic functionality seems to be broken. For instance:

* No panning of graph with either touch or mouse
* No context menu when right clicking on either

The other examples are completely broken, the intial graph does not even render. I tried to update to reflect the changed locations of some .js, and it still did not render.

This to me suggests that not much testing of existing features was done...

**Comment by kristianmandrup** [comment]
The essential (for my needs) functionality works for the network demo.
IMO it needs a complete rewrite from scratch. Ideally much of the functionality extracted and not  tightly integrated with React or any such framework. Yes, a mess mixing Polymer and React his way. Would be better done using custom elements or web components.

I'm fed up with web development tbh. Still working  knee deep in garbage most of the time. Lost my motivation. Can't look at code any more... Need a break for at least 6 months. This code is garbage as well, sorry. Not sure if you are the original author. Can't stand to look at anymore garbage...

**Comment by jonnor** [comment]
Ok, I am glad you were able to massage the code to fit your purposes!
Not a problem that you are not interested in maintaining, I just need to know, since I have to maintain it :)

I am not the original author no, I took over because we use it in flowhub/noflo-ui.
The code might be shit (as is typical), but the code is used and useful, which is more important. More problematic is that there are no tests, and documentation of expected functionality is missing.

I don't believe in rewrites from scratch. Prefer to make it better iteratively. Some desired cleanups are docuemented in #314. Reducing framework dependencies is key. The benefits are small for this usecase, and the costs high due to their invasiveness (Polymer is real bad here) and API churn..

**Comment by kristianmandrup** [comment]
Hey, sorry, was in a bad mood when I replied before ;) I submitted migration notes now. Should be easy to migrate the Polymer classes to Rails classes I think. But yeah, I'm out of the game for a while.. would rather be a farmer than a developer at this point :p

**Comment by jonnor** [comment]
Shit happens :) Thank you for the migration notes!
